api/src/main/java/org/openmrs/module/idgen/validator/AutoGenerationOptionValidator.java

```diff
@@ -0,0 +1,27 @@
+package org.openmrs.module.idgen.validator;
+
+import org.openmrs.module.idgen.AutoGenerationOption;
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+public class AutoGenerationOptionValidator implements Validator {
+
+	@Override
+	public boolean supports(Class<?> clazz) {
+
+		return AutoGenerationOptionValidator.class.isAssignableFrom(clazz);
+	}
+
+	@Override
+	public void validate(Object object, Errors errors) {
+		AutoGenerationOption autoGenaratorionOption = (AutoGenerationOption) object;
+
+		
+		if(!autoGenaratorionOption.isManualEntryEnabled() && autoGenaratorionOption.isAutomaticGenerationEnabled()) {
```

Comment:
@rbuisson   some thing to note here,manual entry is enabled by default,i disabled that while automatic generation is disabled by default,

Reply:
Sorry, it looks like my comment on this got lost or something? Must have made a mistake.
Anyway, what I was explaining is:
We should prevent the user to save an auto-generation option when both options are false, because that simply doesn't make sense to **disable** manual entry **and disable** automatic generation.

It seems that the test you are doing here is checking for manual entry disabled and auto-generation **enabled**, which is of course a possible use case.